nadia/builder_mapping.py

```diff
@@ -0,0 +1,39 @@
+"""Implementation of class responsible for obtaining schema builders for OpenAPI types."""
+import collections
+import logging
+from nadia.array import ArrayBuilder
+from nadia.object import ObjectBuilder
+from nadia.primitives import FloatBuilder, IntegerBuilder, StringBuilder
+
+
+class BuilderMapping(collections.Mapping):
+    """Caching mapping of builders."""
+
+    builder_factories = {
+        'number': FloatBuilder,
+        'integer': IntegerBuilder,
+        'string': StringBuilder,
+        'object': ObjectBuilder,
+        'array': ArrayBuilder}
+
+    def __init__(self, builder_factories=None):
```

Comment:
What I have in mind is possibility of defining custom builders in case one would like to decode data into objects of slightly different types. Take a look at this gist in which Decimal field is used instead of Float field for decoding values of `number` type: https://gist.github.com/dexter2206/9f8faa0fdc60fee8fc4b78c438983995

Even more generally, one could define builders based on custom marshmallow fields. A possible application would be creating a plugin allowing nadia to decode numerical data into numpy arrays automatically.

I would like to keep this functionallity but I agree that implementation might be overcomplicated. Let me know if you have some simpler alternative in mind!